src/clj/puppetlabs/services/protocols/request_handler.clj

```diff
@@ -3,4 +3,7 @@
 (defprotocol RequestHandlerService
   (handle-request
     [this request]
-    "Handles a request from an agent."))
+    "Handles a request from an agent.")
+  (handle-invalid-request
```

Comment:
I view the `request_handler` service as strictly being responsible for requests that require JRuby interaction.  (We should probably have chosen a name that makes that more obvious.)

Could we put the 404 stuff directly in the `master` service?